windows/vtpm_cold_add_remove/bitlocker_enable_test.yml

```diff
@@ -0,0 +1,122 @@
+# Copyright 2024 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Check C: drive used space is encrypted, protection is on,
+# and encryption key protector is TPM after enabling BitLocker
+#
+# From Windows 11 v24H2, hardware requirements for Automatic Device Encryption are:
+# TPM 2.0 and UEFI secure boot,
+# please refer to this doc: https://learn.microsoft.com/en-us/windows-hardware/design/device-experiences/oem-bitlocker
+- name: "Get VM firmware info for Windows client"
+  include_tasks: ../../common/vm_get_boot_info.yml
+  when: guest_os_product_type == 'client'
+
+- name: "Set fact of whether BitLocker Drive Encryption is enabled automatically"
+  ansible.builtin.set_fact:
+    win_bde_auto_enable: "{{ guest_os_product_type == 'client' and guest_os_ansible_distribution_ver is version('10.0.26100.0', '>=') and vm_secureboot_enabled }}"
+
+# BitLocker is not installed by default in Windows Server
+- name: "Check and install BitLocker feature in Windows Server"
+  when: guest_os_product_type == 'server'
+  block:
+    - name: "Check if BitLocker is installed"
+      include_tasks: ../utils/win_execute_cmd.yml
+      vars:
+        win_powershell_cmd: "(Get-WindowsFeature -Name BitLocker).InstallState"
+    - name: "Set fact of BitLocker install state"
+      ansible.builtin.set_fact:
+        win_srv_bitlocker: "{{ win_powershell_cmd_output.stdout_lines[0].strip() }}"
+
+    - name: "BitLocker is already installed"
+      ansible.builtin.debug: var=win_srv_bitlocker
+      when: win_srv_bitlocker in ['Installed', 'Enabled']
+
+    - name: "BitLocker is not installed"
+      when: win_srv_bitlocker not in ['Installed', 'Enabled']     
+      block:
+        - name: "Install BitLocker"
+          include_tasks: ../utils/win_execute_cmd.yml
+          vars:
+            win_powershell_cmd: "Install-WindowsFeature BitLocker"
+        - name: "Restart guest OS after BitLocker install"
+          include_tasks: ../utils/win_shutdown_restart.yml
+          vars:
+            set_win_power_state: "restart"
+
+- name: "Handle BitLocker Device Encryption not enabled automatically situation"
+  when: not win_bde_auto_enable
+  block:
+    - name: "Enable BitLocker on 'C:' drive"
+      include_tasks: ../utils/win_execute_cmd.yml
+      vars:
+        win_powershell_cmd: "Enable-BitLocker -MountPoint 'C:' -UsedSpaceOnly -TpmProtector -SkipHardwareTest"
+
+    - name: "Set fact of encounter known issue on {{ guest_os_ansible_distribution }}"
+      ansible.builtin.set_fact:
+        win_bde_known_issue: true
```

Comment:
This variable is only used once, why not directly include the task `handle_vm_known_issue.yml` at here?

Reply:
> This variable is only used once, why not directly include the task `handle_vm_known_issue.yml` at here?

Since the conditions `win_powershell_cmd_output.stderr` of this `set_fact` is referred to the above task result, it's more clear logically here, when it meets the condition here then handle this known issue. And also tasks in the `handle_vm_known_issue.yml` will not be in a large block.